### PR TITLE
fix-endpoint-sample-metrics

### PR DESCRIPTION
**Pull request by ChrOertlin**
## Description

Replace SampleLaneSequencingMetrics in endpoint

### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b [THIS-BRANCH-NAME] -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/server/api.py [hunk 1]
```diff
@@ -364,21 +364,35 @@ def get_sequencing_metrics(flow_cell_name: str):
     """Return sample lane sequencing metrics for a flow cell."""
     if not flow_cell_name:
```
Comment (diitaz93) [inline cg/server/api.py:365]: change this to `flow_cell_id` to be clear

File: cg/server/api.py [hunk 2]
```diff
@@ -364,21 +364,35 @@ def get_sequencing_metrics(flow_cell_name: str):
...
         )
-    metrics_dtos: list[SequencingMetricsRequest] = [
-        SequencingMetricsRequest.model_validate(metric, from_attributes=True)
-        for metric in sequencing_metrics
-    ]
+    metrics_dtos: list[SequencingMetricsRequest] = parse_metrics_into_request(sequencing_metrics)
     return jsonify([metric.model_dump() for metric in metrics_dtos])
 
 
+def parse_metrics_into_request(
```
Comment (ChrOertlin) [inline cg/server/api.py:379]: not sure where to best out this,... There are some other functions in here

File: cg/server/api.py [hunk 3]
```diff
@@ -364,21 +364,35 @@ def get_sequencing_metrics(flow_cell_name: str):
...
+    metrics: list[IlluminaSampleSequencingMetrics],
+) -> list[SequencingMetricsRequest]:
+    """Parse  sequencing metrics into a request."""
```
Comment (diitaz93) [inline cg/server/api.py:382]: extra space in docstring
```suggestion
    """Parse sequencing metrics into a request."""
```

File: cg/server/dto/sequencing_metrics/sequencing_metrics_request.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
 """DTO for the sequencing metrics."""
 
-from pydantic import BaseModel
+from pydantic import BaseModel, Field
```
Comment (diitaz93) [inline cg/server/dto/sequencing_metrics/sequencing_metrics_request.py:3]: Do you need field?

File: cg/server/dto/sequencing_metrics/sequencing_metrics_request.py [hunk 5]
```diff
@@ -1,6 +1,6 @@
...
 
 
 class SequencingMetricsRequest(BaseModel):
```
Comment (diitaz93) [inline cg/server/dto/sequencing_metrics/sequencing_metrics_request.py:6]: if possible change the `flow_cell_name` attribute  to `flow_cell_id`

**Review by diitaz93 — APPROVED** [review]
Great 👍

File: cg/server/api.py (hunk 2 above)
Comment (diitaz93) [reply cg/server/api.py:379]: maybe create a utils module in `cg/server/`

File: cg/server/dto/sequencing_metrics/sequencing_metrics_request.py (hunk 5 above)
Comment (ChrOertlin) [reply cg/server/dto/sequencing_metrics/sequencing_metrics_request.py:6]: Not possible without a change on the other side.

File: cg/server/api.py (hunk 1 above)
Comment (ChrOertlin) [reply cg/server/api.py:365]: Would change the signature, not sure how that impacts callers..

**Pull request merged**